gobblin-core/src/main/java/gobblin/source/extractor/hadoop/HadoopFsHelper.java

```diff
@@ -0,0 +1,95 @@
+package gobblin.source.extractor.hadoop;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.source.extractor.filebased.FileBasedHelper;
+import gobblin.source.extractor.filebased.FileBasedHelperException;
+import gobblin.util.ProxiedFileSystemWrapper;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class HadoopFsHelper implements FileBasedHelper {
+  private final State state;
+  private final Configuration configuration;
+  private FileSystem fs;
+
+  protected HadoopFsHelper(State state, Configuration configuration) {
+    this.state = state;
+    this.configuration = configuration;
+  }
+
+  protected State getState() {
+    return this.state;
+  }
+
+  public FileSystem getFileSystem() {
+    return this.fs;
+  }
+
+  @Override
+  public void connect() throws FileBasedHelperException {
+    String uri = state.getProp(ConfigurationKeys.SOURCE_FILEBASED_FS_URI);
+    try {
+      if (uri == null) {
+        throw new FileBasedHelperException(ConfigurationKeys.SOURCE_FILEBASED_FS_URI + " has not been configured");
+      }
+      this.createFileSystem(uri);
+    } catch (IOException e) {
+      throw new FileBasedHelperException("Cannot connect to given URI " + uri + " due to " + e.getMessage(), e);
+    } catch (URISyntaxException e) {
+      throw new FileBasedHelperException("Malformed uri " + uri + " due to " + e.getMessage(), e);
+    } catch (InterruptedException e) {
+      throw new FileBasedHelperException("Interrupted exception is caught when getting the proxy file system", e);
+    }
+  }
+
+  @Override
+  public List<String> ls(String path) throws FileBasedHelperException {
+      List<String> results = new ArrayList<>();
```

Comment:
I would prefer to not make this change at the moment.  This code was pulled directly from `AvroFsHelper.java`.  This brings up the broader question as to when changes like this should be made.  My preference would be to minimize the changes, only fixing obvious issues.  If another PR is needed, to address these issues, then so be it.